### ISSUE TITLE
[Bug]: OpenVINO opset MaxPool accepts pad larger than kernel on CPU

### ISSUE THREAD

**Issue by ALinrunrun** [openvinotoolkit/openvino#35759]
### OpenVINO Version

2026.1.0

### Operating System

Other (Please specify in description)

### Device used for inference

CPU

### Framework

None

### Model used

Minimal OpenVINO model generated by the reproducer script.

### Issue description

OpenVINO CPU accepts a `MaxPool` model built directly with the OpenVINO Python opset API where the padding is larger than the kernel size.

For the same configuration, PyTorch rejects the operation with a runtime error, while OpenVINO compiles and runs it, returning an output tensor.

Framework: OpenVINO Python API
OS: Ubuntu 24.04.2 LTS / Linux x86_64

### Step-by-step reproduction

```
#!/usr/bin/env python3
import sys
import numpy as np
import torch
import torch.nn.functional as F
import openvino as ov
from openvino import opset13 as ops, Type

x = np.random.RandomState(11).randn(1, 1, 5, 5).astype(np.float32)

p = ops.parameter([1, 1, 5, 5], dtype=Type.f32)
mp = ops.max_pool(
    p,
    strides=[1, 1],
    dilations=[1, 1],
    pads_begin=[3, 3],
    pads_end=[3, 3],
    kernel_shape=[2, 2],
)

model = ov.Model([mp.output(0)], [p])
ov_out = ov.Core().compile_model(model, "CPU")([x])[0]

torch_err = None
try:
    F.max_pool2d(torch.tensor(x), kernel_size=2, padding=3, stride=1)
except Exception as e:
    torch_err = type(e).__name__

print("torch raised:", torch_err)
print("OpenVINO accepted, shape:", ov_out.shape)

if torch_err is not None and ov_out.size > 0:
    print("BUG REPRODUCED: OpenVINO opset MaxPool accepts pad larger than kernel.")
    sys.exit(0)

print("NOT REPRODUCED")
sys.exit(1)
```

### Relevant log output

```shell
Expected:
OpenVINO CPU should reject MaxPool when the padding is larger than the kernel size, consistently with PyTorch, or document this behavior clearly.

Actual:
torch raised: RuntimeError
OpenVINO accepted, shape: (1, 1, 10, 10)

BUG REPRODUCED: OpenVINO opset MaxPool accepts pad larger than kernel.
```

### Issue submission checklist

- [x] I'm reporting an issue. It's not a question.
- [x] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [x] There is reproducer code and related data files such as images, videos, models, etc.

**Comment by mlukasze** [comment]
Ref. CVS-186545
